### DISCUSSION TITLE
cy.origin() Cypress not opening page from another domain

### DISCUSSION THREAD

**Discussion by kbodya** [cypress-io/cypress#26014] [Questions and Help]
H everyone. Please help me to resolve the following problem 

On my project after clicking on the "MakePayment" button, the user is redirected to the payment page. URL of the payment page looks like this  "https://checkout.stripe.com/c/pay/cs_test_a1E8WaWpykZA5UJXYg81r3YzULc1a1nnnEHsIOn" 
where 
"cs_test_a1E8WaWpykZA5UJXYg81r3YzULc1a1nnnEHsIOn" is unique for each user and generated new every time after clicking the "MakePayment" button

The problem is that after the user goes to the page https://checkout.stripe.com/c/pay/cs_test_a1E8WaWpykZA5UJXYg81r3YzULc1a1nnnEHsIOn 
its content is not displayed

>  cy.OpenMakePaymentTest()
    cy.get (makePayment.radtioCustomAmount).click()
    cy.get (makePayment.textInputField).type (100)
    .invoke('removeAttr', 'iframe')
    cy.get (makePayment.makePaymentButton).click()
    cy.wait (6000)
    cy.origin ('https://checkout.stripe.com/c/pay', () => {
    })
> 
**Cypress 12.7.0
Stripe payment system**

Video capture with Cypress test - https://drive.google.com/file/d/1Tgi6bvLWfnVGzsGEfTmmVoIu_B9TG5Zb/view?usp=share_link

**Comment by portyu9** [comment]
Hi @kbodya 

`cy.origin()` does **not navigate the browser to that origin**. It tells Cypress that the commands inside its callback should run against that secondary origin.

So this:

```js
cy.origin('https://checkout.stripe.com/c/pay', () => {})
```

has an empty callback and doesn't cause Stripe Checkout to load.

The navigation should happen because of your application's payment button. After that, put the Cypress commands that operate on the Stripe origin inside `cy.origin()`:

```js
cy.get(makePayment.makePaymentButton).click()

cy.origin('https://checkout.stripe.com', () => {
  cy.location('hostname').should('eq', 'checkout.stripe.com')
  cy.get('body').should('be.visible')
})
```

Also, this line:

```js
.invoke('removeAttr', 'iframe')
```

does not remove or bypass an iframe. It literally removes an HTML attribute named `iframe` from the selected element.

There is an additional distinction if the Stripe Checkout page itself contains payment UI inside cross-origin iframes: `cy.origin()` handles interaction with a **different top-level origin**, but it is not a general solution for interacting inside arbitrary cross-origin iframe contents.

For Stripe specifically, I would keep the browser test focused on verifying that your application creates the checkout session and redirects to the expected Stripe Checkout page, and use Stripe's supported test/API mechanisms for payment scenarios where possible.

So first remove the fixed `cy.wait(6000)`, let the button initiate the navigation, and put actual assertions/interactions inside the `cy.origin('https://checkout.stripe.com', ...)` callback.

Please mark this answer as accepted if it helped, thank you.
